pkg/commands/test/test.go

```diff
@@ -325,7 +336,18 @@ func runQuery(ctx context.Context, query string, input interface{}, compiler *as
 			value := e.Value
 			if hasResults(value) {
 				for _, v := range value.([]interface{}) {
-					errs = append(errs, errors.New(v.(string)))
+					switch val := v.(type) {
+					case string:
+						errs = append(errs, errors.New(val))
+					case map[string]interface{}:
+						if _, ok := val["msg"]; !ok {
+							panic("invalid rule violation format (%v), msg needs to be set")
```

Comment:
instead of a `panic` could we return an error upstream? I think in this case of a bad policy we should allow the program to continue and process other violations

Reply:
I agree, I just used a panic for now as that mirrors the current behavior if you provide an incorrect type. Will look into improving this